elf/module.go

```diff
@@ -770,3 +786,56 @@ func (b *Module) CloseExt(options map[string]CloseOptions) error {
 	}
 	return nil
 }
+
+// UpdateSecName takes a mapping type which represents one of the mappings that current module stores,
+// and update the name on the section object from "oldName" to "newName".
+// Note that this doesn't change the key in the mapping
+func (b *Module) UpdateSecName(mappingType MappingType, oldName, newName string) error {
+	found := false
+
+	switch mappingType {
```

Comment:
This switch could just determine which map you use, and below you can actually call it.

e.g.

```
...
  switch mappingType {
    case TypeMap:
      m = b.maps
    case TypeProbe:
      m = b.probes
    ...
  }

  if sec, ok := m[oldName]; ok {
    sec.Name = newname
    return nil
  }

  return fmt.Errorf("no such section \"%s\" exists in mapping %v", oldName, mappingType)
}